frontend/src/Frontend/UI/Wallet.hs

```diff
@@ -122,117 +125,79 @@ uiAvailableKeys aWallet = do
+  switchHold never <=< dyn $ ffor md $ \case
+    Nothing -> pure never
+    Just account -> elClass "tr" "table__row" $ do
+      el "td" $ dynText $ unAccountName . _account_name <$> account
+
+      let public = keyToText . _keyPair_publicKey . _account_key <$> account
       elClass "td" "wallet__key wallet__key_type_public" $ dynText public
-      el "td" $
-        void $ copyButton copyBtnCfg $ current public
-
-      let private = maybe "No key" keyToText . _keyPair_privateKey <$> k
-      isShown <- keyCopyWidget "td" "wallet__key wallet__key_type_private" private
-      el "td" $ do
-        let cfg = copyBtnCfg & uiButtonCfg_disabled .~ fmap not isShown
-        void $ copyButton cfg $ current private
+      el "td" $ dynText $ Pact._chainId . _account_chainId <$> account
+      el "td" $ dynText $ _account_notes <$> account
+      -- TODO balance
+      el "td" $ text "Balance Placeholder"
 
       onDel <- elClass "td" "wallet__delete" $
         deleteButton $
           def & uiButtonCfg_title .~ Just "Delete key permanently"
               & uiButtonCfg_class %~ (<> "wallet__add-delete-button")
 
-      pure (const n <$> onDel)
-  where
-    copyBtnCfg =
-      btnCfgTertiary
-        & uiButtonCfg_class %~ (fmap (<> "button_size_tiny"))
-        & uiButtonCfg_title .~ pure (Just "Copy key to clipboard")
-
--- | Widget showing/hiding a private key.
---
---   The returned Dynamic tells whether the key is currently hidden or shown.
-keyCopyWidget
-  :: MonadWidget t m
-  => Text
-  -> Text
-  -> Dynamic t Text
-  -> m (Dynamic t Bool)
-keyCopyWidget t cls keyText = mdo
-  isShown <- foldDyn (const not) False (domEvent Click e)
-  let mkShownCls True = ""
-      mkShownCls False = " wallet__key_hidden"
-
-  (e, _) <- elDynClass t (pure cls <> fmap mkShownCls isShown) $ do
-    let
-      mkText True k = k
-      mkText False _ = "****************************"
-
-    elDynClass' "span" "key-content" $
-      dynText (mkText <$> isShown <*> keyText)
-  pure isShown
+      pure (attachWith (\a _ -> (i, _account_name a)) (current account) onDel)
 
 -- | Get the balance of an account from the network. 'Nothing' indicates _some_
 -- failure, either a missing account or connectivity failure. We have no need to
 -- distinguish between the two at this point.
-getBalance :: (MonadWidget t m, HasNetwork model t) => model -> ChainId -> Event t AccountName -> m (Event t (Maybe Decimal))
+getBalance
+  :: (MonadWidget t m, HasNetwork model t, HasCrypto key (Performable m))
+  => model -> ChainId -> Event t AccountName -> m (Event t (Maybe AccountBalance))
 getBalance model chain account = do
   networkRequest <- performEvent $ attachWith mkReq (current $ getNetworkNameAndMeta model) account
   response <- performLocalReadCustom (model ^. network) pure networkRequest
-  let toBalance (_, [Right (_, Pact.PLiteral (Pact.LDecimal d))]) = Just d
+  let toBalance (_, [Right (_, Pact.PLiteral (Pact.LDecimal d))]) = Just $ AccountBalance d
```

Comment:
We should probably see if Pact has `Prism`s for this, that could be my preference leaking in though. 

```
toBalance = preview (_2 . _head . _Right . _2 . _PLiteral . _PDecimal. _Just . to AccountBalance)
```

Reply:
I don't think they do